libraries/rush-sdk/src/RushSdkLoader.ts

```diff
@@ -0,0 +1,248 @@
+import { SpawnSyncReturns } from 'child_process';
+
+const RUSH_LIB_NAME: string = '@microsoft/rush-lib';
+
+/**
+ * @internal
+ */
+export type _RushLibModuleType = Record<string, unknown>;
+
+/**
+ * @beta
+ */
+export interface IRushSdkLoaderOptions {
+  terminal?: ITerminal;
+}
+
+declare const global: NodeJS.Global &
+  typeof globalThis & {
+    ___rush___rushLibModule?: _RushLibModuleType;
+    ___rush___rushLibModuleFromInstallAndRunRush?: _RushLibModuleType;
+  };
+
+/**
+ * @beta
+ */
+export class RushSdkLoader {
+  /**
+   * @internal
+   */
+  public static _rushLibModule: _RushLibModuleType | undefined = RushSdkLoader._tryLoadRushLib();
+
+  /**
+   * @internal
+   */
+  public static _rushLibModuleHasBeenInstalled: boolean = !!RushSdkLoader._rushLibModule;
+
+  public static install(options?: IRushSdkLoaderOptions): void {
+    if (RushSdkLoader._rushLibModuleHasBeenInstalled) {
+      return;
+    }
+
+    const { terminal = new Terminal(new ConsoleTerminalProvider()) } = options || {};
+
+    // SCENARIO 3:  A tool or script depends on "rush-sdk", and is meant to be used inside a monorepo folder.
+    // In this case, we can use install-run-rush.js to obtain the appropriate rush-lib version for the monorepo.
+    if (RushSdkLoader._rushLibModule === undefined) {
+      try {
+        const rushJsonPath: string | undefined = RushSdkLoader._tryFindRushJsonLocation(process.cwd());
+        if (!rushJsonPath) {
+          throw new Error(
+            'Unable to find rush.json in the current folder or its parent folders.\n' +
+              'This tool is meant to be invoked from a working directory inside a Rush repository.'
+          );
+        }
+        const monorepoRoot: string = path.dirname(rushJsonPath);
+
+        const rushJson: JsonObject = JsonFile.load(rushJsonPath);
+        const { rushVersion } = rushJson;
+
+        const installRunNodeModuleFolder: string = `${monorepoRoot}common/temp/install-run/@microsoft+rush@${rushVersion}`;
+        try {
+          // First, try to load the version of "rush-lib" that was installed by install-run-rush.js
+          terminal.writeVerboseLine(`Trying to load  ${RUSH_LIB_NAME} installed by install-run-rush`);
+          RushSdkLoader._rushLibModule =
+            RushSdkLoader._requireRushLibUnderFolderPath(installRunNodeModuleFolder);
+        } catch (e) {
+          let installAndRunRushStderrContent: string = '';
+          try {
+            const installAndRunRushJSPath: string = path.join(
+              monorepoRoot,
+              'common/scripts/install-run-rush.js'
+            );
+
+            terminal.writeLine('The Rush engine has not been installed yet. Invoking install-run-rush.js...');
+
+            const installAndRuhRushProcess: SpawnSyncReturns<string> = Executable.spawnSync('node', [
+              installAndRunRushJSPath,
+              '--help'
+            ]);
+            terminal.write(installAndRuhRushProcess.stdout);
```

Comment:
Note: `rush --help` lists all rush commands information, but the idea is just trigger the `install-run` logic behind. I can see this stdout would be pretty annoying, so it's intentionaly removed in original implementation.